### DISCUSSION TITLE
ts-node-dev with next.js-10.0.8 TypeError: Unexpected response from worker: undefined

### DISCUSSION THREAD

**Discussion by Fi1osof** [vercel/next.js#23070] [Help]
### What version of Next.js are you using?

10.0.8

### What version of Node.js are you using?

15.6.0

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu-20

### How are you deploying your application?

cross-env NODE_ENV=development TS_NODE_PROJECT=tsconfig.server.json ts-node-dev --ignore-watch .next --rs --no-notify --respawn --transpile-only server

### Describe the Bug

```
$ cross-env NODE_ENV=development TS_NODE_PROJECT=tsconfig.server.json ts-node-dev --ignore-watch .next --rs --no-notify --respawn --transpile-only server
[INFO] 12:14:25 ts-node-dev ver. 1.1.6 (using ts-node ver. 9.1.1, typescript ver. 4.2.3)
[HPM] Proxy created: /  -> https://api.prisma-cms.com
[HPM] Proxy rewrite rule created: "^/api(/|$)" ~> "/"
TypeError: Unexpected response from worker: undefined
    at ChildProcessWorker.onMessage (/disk480/www/nwb/@prisma-cms/nextjs/node_modules/next/node_modules/jest-worker/build/workers/ChildProcessWorker.js:257:15)
    at ChildProcess.emit (node:events:391:22)
    at emit (node:internal/child_process:918:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
[ERROR] 12:14:25 TypeError: Unexpected response from worker: undefined

```

### Expected Behavior

Start well

### To Reproduce

Repo: https://github.com/prisma-cms/nextjs/tree/ts-node-dev-error

On next.js <= 10.0.7 works fine.

**Comment by portenez** [comment]
I'm having the same problem. There's probably a file that gets included in a weird way that `ts-node` has to include manually.

**Reply by Fi1osof** [reply]
I think in my case this due use ts-node-dev with different jest-worker version.
https://github.com/prisma-cms/nextjs/blob/cb1d045a10db2812ddbb6cd0145e9cd905bd2c67/package.json#L8

next dev works fine.

**Comment by oemilsson** [comment]
You found any solution for this? Im stuck with the same issue and i cant use next dev since im having a custom server.

**Reply by Fi1osof** [reply]
@oemilsson no. Or next@10.0.7 + ts-node-dev, or yarn next dev with next@latest.
```
    "dev": "next dev",
    "ts-dev": "cross-env NODE_ENV=development TS_NODE_PROJECT=tsconfig.server.json ts-node-dev --ignore-watch .next --rs --no-notify --respawn --transpile-only server",
```

**Comment by portenez** [comment]
We switched to ts-node and nodemon. Works fine this way.

**Reply by Fi1osof** [reply]
> We switched to ts-node and nodemon. Works fine this way.

This is my project with next@latest: https://github.com/prisma-cms/nextjs/tree/ts-node-dev-error

try to run `yarn ts-dev`

If i delete next.config.js, all works fine. But if i create next.config.js even with default code 
```js
module.exports = {
  /* config options here */
}
```

i got error
```
TypeError: Unexpected response from worker: undefined
    at ChildProcessWorker._onMessage (/.../next/node_modules/jest-worker/build/workers/ChildProcessWorker.js:264:15)
    at ChildProcess.emit (node:events:381:22)
    at emit (node:internal/child_process:920:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
[ERROR] 17:34:14 TypeError: Unexpected response from worker: undefined
^C

```

**Comment by DanielOrtel** [comment]
FYI, having the same issue, and I'm not using TS. 

```
/Users/[...]/node_modules/next/node_modules/jest-worker/build/workers/ChildProcessWorker.js:264
        throw new TypeError('Unexpected response from worker: ' + response[0]);
        ^

TypeError: Unexpected response from worker: undefined
    at ChildProcessWorker._onMessage (/Users/[...]/node_modules/next/node_modules/jest-worker/build/workers/ChildProcessWorker.js:264:15)
    at ChildProcess.emit (events.js:314:20)
    at emit (internal/child_process.js:877:12)
    at processTicksAndRejections (internal/process/task_queues.js:85:21)
undefined
```

**Reply by DanielOrtel** [reply]
~~Also, shouldn't this be a bug issue? Looks to me like it's a bug with Next.js~~ Looks to be a conflict with nodemon(or a version of it), dunno if that constitutes a bug or not. 

Specifically, our repo was fine until `next` version 10.0.7. But if we upgrade to 10.0.8, it starts to happen already. Currently we're trying to bump `nextjs` to it's latest release, but can't because of this. 

Happens on `next dev` and `next build` as well.

**Comment by DanielOrtel** [comment]
Found the issue with my problem. I have a package which contains the webpack configs for all the apps in the monorepo, and it exposes multiple different webpack configs. One of those configs required `nodemon`, and for that reason the whole thing crashed. If I removed the `require('nodemon')` bit, the next server started up just fine. So my best guess, is that this happens due to `nodemon` being required somewhere for the other people as well.

**Reply by langri-sha** [reply]
Same here! Exporting all webpack plugins from a single package. Yoinked out the `nodemon-webpack-plugin` export and everything works now.

**Comment by Fi1osof** [comment]
Possible answer here: https://github.com/wclr/ts-node-dev/issues/264#issuecomment-842992756

**Comment by latchy** [comment]
Was there ever a concrete solution found?
I'm still having this issue.

**Reply by Fi1osof** [reply]
I have too. Know no solution.

**Comment by KennethWussmann** [comment]
The issue still exists with Next.JS 13. 
Adding this experimental flag to the `next.config.js` fixed it for me:

```javascript
const nextConfig = {
  // ...
  experimental: {
    workerThreads: true,
  },
};
```

**Reply by jamoy** [reply]
If anyone ends up here having problems with catchall routes `app/[id]/page.jsx` with nested fetches in their layouts in a big next 14 codebase. this silenced the hundreds of

```
TypeError: Unexpected response from worker: undefined
```

errors for me

**Reply by nerdyman** [reply]
This also fixed the issue for me in Next 16.
